### ISSUE TITLE
Add support for LR schedulers using validation metrics

### ISSUE THREAD

**Issue by arunppsg** [deepchem/deepchem#3562]
Some learning rate scheduler like [ReduceLROnPlateau](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html) monitors a metric to update the learning rate scheduler. The metric is passed along the `step` method of learning rate scheduler. We need to add support for such dynamic learning rate schedulers as they are used in implementation of models like Infomax3D.
